CHANGELOG.md

```diff
@@ -1,5 +1,198 @@
+
+Starting with v1.0.0, input script files are **never** validated. This means that TestCafe executes all the scripts you specify as test sources. If you use Glob patterns to specify input test files, please recheck these patterns to avoid unintended file matches.
+
+The `--disable-test-syntax-validation` command line flag and the `disableTestSyntaxValidation` option for the [runner.run](https://devexpress.github.io/testcafe/documentation/using-testcafe/programming-interface/runner.html#run) API method that disabled test syntax validation were removed in v1.0.0.
+
+#### :boom: Custom Request Hooks: Asynchronous API
+
+[Request hook](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/) methods became asynchronous in TestCafe v1.0.0.
+
+If you use a [custom request hook](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html), add the `async` keyword before the [onRequest](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html#the-onrequest-method) and [onResponse](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html#the-onresponse-method) methods.
+
+```js
+import { RequestHook } from 'testcafe';
+class MyRequestHook extends RequestHook {
+    constructor (requestFilterRules, responseEventConfigureOpts) {
+        super(requestFilterRules, responseEventConfigureOpts);
+        // ...
+    }
+    async onRequest (event) {
+        // ...
+    }
+    async onResponse (event) {
+        // ...
+    }
+}
+```
+
+#### :boom: Custom Reporter Plugins: Asynchronous API
+
+TestCafe v1.0.0 also introduces asynchronous API for [reporter plugins](https://devexpress.github.io/testcafe/documentation/extending-testcafe/reporter-plugin/).
+
+If you maintain a custom reporter plugin, add the `async` keyword before each reporter method:
+
+* [reportTaskStart](https://devexpress.github.io/testcafe/documentation/extending-testcafe/reporter-plugin/reporter-methods.html#reporttaskstart)
+* [reportFixtureStart](https://devexpress.github.io/testcafe/documentation/extending-testcafe/reporter-plugin/reporter-methods.html#reportfixturestart)
+* [reportTestDone](https://devexpress.github.io/testcafe/documentation/extending-testcafe/reporter-plugin/reporter-methods.html#reporttestdone)
+* [reportTaskDone](https://devexpress.github.io/testcafe/documentation/extending-testcafe/reporter-plugin/reporter-methods.html#reporttaskdone)
+
+```js
+async reportTaskStart (startTime, userAgents, testCount) {
+    // ...
+},
+async reportFixtureStart (name, path, meta) {
+    // ...
+},
+async reportTestDone (name, testRunInfo, meta) {
+    // ...
+},
+async reportTaskDone (endTime, passed, warnings, result) {
+    // ...
+}
+```
+
+#### :boom: Programming Interface: Multiple Method Calls Prohibited
```

Comment:
It needs some motivation IMO.

Reply:
I'll copy motivation from here once it's approved:
https://github.com/DevExpress/testcafe/pull/3388/files#diff-1ce235e72b3786322b3f4b017ce27cd7R80

Will it be enough?